### PR TITLE
ENH: create and attach ISM policy

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*
#52 

*Description of changes:*
This PR enhances the ES sink to

(1) check and create ISM policy for raw trace span if ISM plugin is enabled
(2) attach ISM policy to index template

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java [hunk 1]
```diff
@@ -153,17 +166,53 @@ public void stop() {
...
     final URL jsonURL = esSinkConfig.getIndexConfiguration().getTemplateURL();
-    final String templateJson = readTemplateURL(jsonURL);
-    putIndexTemplateRequest.source(templateJson, XContentType.JSON);
+    final Map<String, Object> template = readTemplateURL(jsonURL);
+    if (ismEnabled && isRaw) {
+      // Attach policy_id and rollover_alias
+      final Map<String, Object> settings = (Map<String, Object>) template.getOrDefault("settings", new HashMap<>());
+      settings.put(IndexConstants.ISM_POLICY_ID_SETTING, IndexConstants.RAW_ISM_POLICY);
+      settings.put(IndexConstants.ISM_ROLLOVER_ALIAS_SETTING, indexAlias);
+    }
```
[the remark this answers is not in the document]
Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java:214]: Sorry, not following here. What do you mean by "Can we not instead create index"?

File: situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java [hunk 2]
```diff
@@ -68,8 +77,12 @@ public ElasticsearchSink(final PluginSetting pluginSetting) {
 
   public void start() throws IOException {
     restHighLevelClient = esSinkConfig.getConnectionConfiguration().createClient();
+    final boolean ismEnabled = checkISMEnabled();
+    if (ismEnabled) {
+      checkAndCreatePolicy();
+    }
```
[the remark this answers is not in the document]
Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java:83]: Yes.

File: situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java]: Discussed offline

**Review by kowshikn — APPROVED** [review]
👍🏽

**Comment by kowshikn** [comment]
The ISM integ tests will be done separately right?

**Comment by chenqi0805** [comment]
> The ISM integ tests will be done separately right?

Yes. In the pending PR

**Comment by chenqi0805** [comment]
NOTE: I use my admin privilege to merge this one to unblock myself.

**Pull request merged**